### PR TITLE
Initial testing additions for spin up of chart

### PR DESCRIPTION
**Pull request by pcm32**
This PR adds logic for spinning up the chart as part of the tests. It is based on an older setup, so possibly some additional things could be removed.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,24 +1,56 @@
...
+  - kubectl cluster-info
+  - helm dependency update ./galaxy
+  - helm install -f helm-configs/.travis-deployment.yaml --name travis ./galaxy
+  # Check charts format in the meantime.
   - >
      for dir in galaxy; do
       helm lint $dir
-      if [ $? != 0 ]; then
-       travis_terminate 1
-      fi
```
Comment (pcm32) [inline .travis.yml]: actually, I think that we should leave this... no point in running things if the linting fails.

File: .travis.yml [hunk 2]
```diff
@@ -1,24 +1,59 @@
...
 install:
   # Installing Helm
   - wget -q ${HELM_URL}/${HELM_TGZ}
   - tar xzfv ${HELM_TGZ}
   - PATH=`pwd`/linux-amd64/:$PATH
-  - helm init --client-only
+
+before_script:
+  - sudo mount --make-rshared /
+  - sudo apt-get install -y socat
```
Comment (nuwang) [inline .travis.yml]: I don't see socat being called anywhere. Is this a minikube dependency?

File: .travis.yml [hunk 3]
```diff
@@ -1,24 +1,59 @@
...
      done
+  # Describe pods as they are sent to run.
+  - >
+     for pod in $(kubectl get pods | awk '{ print $1 }' | grep -v postgres | grep galaxy); do
+      kubectl describe pods/$pod
+     done
+  - bash util/wait-for-deployment -n default -t 600 $(kubectl get deployment | awk '{ print $1 }' | grep 'galaxy-web$')
+  # Show pod logs once they are running
+  - >
+     for pod in $(kubectl get pods | awk '{ print $1 }' | grep -v postgres | grep galaxy); do
```
Comment (nuwang) [inline .travis.yml]: I feel like showing the pod logs is not necessary unless there's an error.

File: .travis.yml [hunk 4]
```diff
@@ -1,24 +1,59 @@
...
+      kubectl logs $pod
+     done
+  # Describe pods as they are running.
+  - >
+     for pod in $(kubectl get pods | awk '{ print $1 }' | grep -v postgres | grep galaxy); do
+      kubectl describe pods/$pod
+     done
+  - kubectl get pods
+  - kubectl get deployments
```
Comment (nuwang) [inline .travis.yml]: Was there a reason to prefer deployment rollout status over polling a Galaxy end point, and maybe running a small test job? The latter would encompass the former?

**Comment by pcm32** [comment]
Once you approve the PR I can squash and merge.

**Comment by pcm32** [comment]
I think that in general we should do that for all our PR merges from now on, gives a cleaner history. I do that in most of our relevant repos.

File: .travis.yml (hunk 4 above)
Comment (pcm32) [reply .travis.yml]: Simply that I copied and adapted an older setup that I had at another repo. There is certainly room for improvement, like what you propose which would be great to have, hopefully in subsequent PRs :-).

File: .travis.yml (hunk 2 above)
Comment (pcm32) [reply .travis.yml]: well, I haven't tried it now, but it used to be due to an issue between helm and minikube. Not sure on current versions.

**Comment by pcm32** [comment]
I think I have addressed most of your very good suggestions @nuwang, if you're happy with this I can squash-merge. I think it would be great to add a tool execution later.

**Comment by almahmoud** [comment]
Actually, would it make sense to add a CVMFS-enabled test here as well?

**Comment by pcm32** [comment]
Oops... missed your comment @almahmoud.... sure, we could do that eventually. We could maybe have some parallel branching tests for different use cases.

**Comment by nuwang** [comment]
> I think that in general we should do that for all our PR merges from now on, gives a cleaner history. I do that in most of our relevant repos.

The best option for a clean history is an interactive rebase. Squashed merges only make sense when the PR has related changes. It wouldn't make sense to squash merge an entire version change for example, since it contains multiple unrelated commits, which makes a single commit impossible to comprehend in isolation. Deciding on a case by case basis is best I think.

**Comment by pcm32** [comment]
> Squashed merges only make sense when the PR has related changes.
This is what should happen most of the times. Otherwise, you're being naughty in the PR. Release/version changes branches are completely different things, but for most feature/bug fixes PRs, I would expect squashed merges.

**Pull request merged**